cpp/src/parquet/column_reader.cc

```diff
@@ -1045,6 +1052,7 @@ class TypedColumnReaderImpl : public TypedColumnReader<DType>,
 
     // Not present for non-repeated fields
     if (this->max_rep_level_ > 0 && rep_levels != nullptr) {
+      DCHECK_GT(this->max_rep_level_, 0);
```

Comment:
Are we gauranteed this directly by line 1054?

Reply:
Indeed, let's remove this line.